File: php/src/Items/ConjuredItem.php
<?php

declare(strict_types=1);

namespace GildedRose\Items;

use GildedRose\Item;
use GildedRose\Items\Abstract\AbstractItem;
use GildedRose\Items\Interface\ItemInterface;

class ConjuredItem extends AbstractItem implements ItemInterface
{
    public function updateQuality(Item $item): void
    {
        // "Conjured" items degrade in Quality twice as fast as normal items
        $this->decreaseQuality($item, $this->getQualityDecrease($item) * 2);
    }
}

File: ruby/items/conjured_item.rb
require_relative 'normal_item'

class ConjuredItem< NormalItem
  def spend_day_in_shop
    decrease_sell_in_day

    decrease_quality
    decrease_quality

    if item.sell_in.negative?
      decrease_quality
      decrease_quality
    end

    keep_quality_lower_limit_in_bounds

    self
  end
end
